test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs

```diff
@@ -2715,26 +2713,29 @@ public void PackCommand_NoWarn_SuppressesWarnings(string property, string value,
                     var xml = XDocument.Load(stream);
 
                     ProjectFileUtils.AddProperty(xml, property, value);
+                    ProjectFileUtils.AddProperty(xml, "PackageLicenseUrl", "http://contoso.com/license.html");
                     ProjectFileUtils.WriteXmlToFile(xml, stream);
                 }
 
                 msbuildFixture.RestoreProject(workingDirectory, projectName, string.Empty);
 
-                var nupkgPath = Path.Combine(workingDirectory, $"{projectName}.1.0.0-rtm.nupkg");
-                var nuspecPath = Path.Combine(workingDirectory, "obj", $"{projectName}.1.0.0-rtm.nuspec");
+                var nupkgPath = Path.Combine(workingDirectory, $"{projectName}.1.0.0.nupkg");
+                var nuspecPath = Path.Combine(workingDirectory, "obj", $"{projectName}.1.0.0.nuspec");
 
-                var result = msbuildFixture.PackProject(workingDirectory, projectName, $"/p:PackageOutputPath={workingDirectory} /p:Version={semver2Version}");
+                var result = msbuildFixture.PackProject(workingDirectory, projectName, $"/p:PackageOutputPath={workingDirectory}");
 
                 Assert.True(File.Exists(nupkgPath), "The output .nupkg is not in the expected place");
                 Assert.True(File.Exists(nuspecPath), "The intermediate nuspec file is not in the expected place");
-                var expectedWarning = string.Format("warning " + NuGetLogCode.NU5105 + ": " + NuGet.Packaging.Rules.AnalysisResources.LegacyVersionWarning, semver2Version);
-                Assert.Equal(result.AllOutput.Contains(expectedWarning), expectToWarn);
-
+                var expectedWarning = string.Format("warning " + NuGetLogCode.NU5125 + ": " + NuGet.Packaging.Rules.AnalysisResources.LicenseUrlDeprecationWarning);
+                if (expectToWarn)
+                {
+                    result.AllOutput.Should().Contain(expectedWarning);
+                }
```

Comment:
This has a different semantic meaning. Previously, when the message was not expected, it asserted that the message was not presented. Now, it only checks that the message is present when it's expected, but does not validate when the message is not expected.

Reply:
yup, nice catch :)

Reply:
fixed.